ui/ui-components/contexts/page.tsx

```diff
@@ -0,0 +1,15 @@
+import { createContext, useContext } from "react";
+import { Actions } from "../utils/apiData";
+
+export interface PageContext {
+  clusterName: Actions.NavigationList["clusterName"];
+  currentTeamMember: Partial<Actions.NavigationList["teamMember"]>;
+  hydrationError?: string;
+  navigation: Actions.NavigationList["navigation"];
+  navigationMode: Actions.NavigationList["navigationMode"];
+  navigationModel: Actions.NavigationList["navigationModel"];
+}
+
+export const PageContext = createContext<PageContext>(undefined);
+
+export const usePageContext = () => useContext(PageContext);
```

Comment:
I'm looking for better suggestions for the name of this context. The other I had in mind was `WebPageContext`.

Reply:
Could call it NavigationContext?